eth/state_accessor.go

```diff
@@ -35,7 +35,17 @@ import (
 // are attempted to be reexecuted to generate the desired state. The optional
 // base layer statedb can be passed then it's regarded as the statedb of the
 // parent block.
-func (eth *Ethereum) stateAtBlock(block *types.Block, reexec uint64, base *state.StateDB, checkLive bool) (statedb *state.StateDB, err error) {
+// Parameters:
+// - block: The block for which we want the state (== state at the stateRoot of the parent)
```

Comment:
@holiman `== state at the stateRoot of the parent` I'm a bit confused by this comment. I feel like what this function does is give you the state at the stateRoot of the block itself?

Reply:
This method returns the `block.Root` yes. 
So if you want to execute block `N`, you need the `stateAt( N-1)`. 
So you need to call this method with `N-1`. 
So the comment should have said

``
// - block: the block to look up the stateRoot of. This needs to be the parent of the block you want to execute
```

Reply:
Ah gotcha thanks